osu.Game.Rulesets.Osu/Edit/Blueprints/HitCircles/Components/HitCircleOverlapMarker.cs

```diff
@@ -92,6 +89,10 @@ public override void UpdateFrom(HitCircle hitObject)
 
                 ring.Scale = new Vector2(1 + 0.1f * ringScale);
                 content.Alpha = 0.9f * (1 - alpha);
+
+                // TODO: should only update colour on skin/combo/object change.
```

Comment:
This is a quick band-aid fix. We are still performing skin lookup on each frame, but now only for visible markers (previously for every alive one). Reworking this will require a bit of refactoring, which should probably be in it's own pr.